haystack/pipelines/utils.py

```diff
@@ -250,31 +254,37 @@ def print_eval_report(
         n_wrong_examples=n_wrong_examples,
         document_scope=document_scope,
         answer_scope=answer_scope,
+        wrong_examples_fields=wrong_examples_fields,
+        max_chars=max_characters_per_wrong_examples_report,
     )
 
     print(f"{pipeline_overview}\n" f"{wrong_examples_report}")
 
 
-def _format_document_answer(document_or_answer: dict):
-    return "\n \t".join(f"{name}: {value}" for name, value in document_or_answer.items())
+def _format_document_answer(document_or_answer: dict, field_filter: List[str] = None):
+    if field_filter is None or len(field_filter) == 0:
+        field_filter = document_or_answer.keys()  # type: ignore
+    return "\n \t".join(f"{name}: {value}" for name, value in document_or_answer.items() if name in field_filter)  # type: ignore
 
 
-def _format_wrong_example(query: dict):
+def _format_wrong_example(
+    query: dict, max_characters_per_wrong_examples_report: int = None, field_filter: List[str] = None
+):
     metrics = "\n \t".join(f"{name}: {value}" for name, value in query["metrics"].items())
-    documents = "\n\n \t".join(map(_format_document_answer, query.get("documents", [])))
+    documents = "\n\n \t".join(_format_document_answer(doc, field_filter) for doc in query.get("documents", []))
     documents = f"Documents: \n \t{documents}\n" if len(documents) > 0 else ""
-    answers = "\n\n \t".join(map(_format_document_answer, query.get("answers", [])))
+    answers = "\n\n \t".join(_format_document_answer(doc, field_filter) for doc in query.get("answers", []))
     answers = f"Answers: \n \t{answers}\n" if len(answers) > 0 else ""
     gold_document_ids = "\n \t".join(query["gold_document_ids"])
     gold_answers = "\n \t".join(query.get("gold_answers", []))
     gold_answers = f"Gold Answers: \n \t{gold_answers}\n" if len(gold_answers) > 0 else ""
     s = (
         f"Query: \n \t{query['query']}\n"
-        f"{gold_answers}"
-        f"Gold Document Ids: \n \t{gold_document_ids}\n"
-        f"Metrics: \n \t{metrics}\n"
-        f"{answers}"
-        f"{documents}"
+        f"{gold_answers[:max_characters_per_wrong_examples_report]}\n"
+        f"Gold Document Ids: \n \t{gold_document_ids[:max_characters_per_wrong_examples_report]}\n"
+        f"Metrics: \n \t{metrics[:max_characters_per_wrong_examples_report]}\n"
+        f"{answers[:max_characters_per_wrong_examples_report]}\n"
+        f"{documents[:max_characters_per_wrong_examples_report]}\n"
```

Comment:
We don't need the cutting here. Lets just cut the individual fields of `answers` and `documents`.